adagrams/game.py

```diff
@@ -1,13 +1,152 @@
 from random import randint
+    # from tests.test_wave_01 import LETTER_POOL
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
 
 def draw_letters():
-    pass
+    letter_pool_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_pool_list.extend([letter] * count)
+        # transfer the dictionary to a 98 letters list to use random.randint
+
+    draw_letter_pool = letter_pool_list.copy()
+    # make a copy list for poping letter to avoid draw a letter and hands repeatly
+    # use index / .pop() use index
+
+    draw_hand = []
+
+    while len(draw_hand) < 10 :
+        random_index = randint(0 ,len(letter_pool_list.copy())-1)
+        draw_letter = letter_pool_list.copy()[random_index]
+        draw_hand.append(letter_pool_list.pop(random_index))
+
+    return draw_hand
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # letters = ["D", "O", "G", "X", "X", "X", "X", "X", "X", "X"]
+    #     word = "DOG"
+    # convert it to dictionaries to mapping the word and letters
+    uppercase_word = word.upper()
+    check_word = {}
+
+    for letter in uppercase_word:
+        if letter in check_word:
+            check_word[letter] += 1
+        else:
+            check_word[letter] = 1
```

Comment:
I love your idea to create frequency dictionaries for the word and the letter bank! This is less of a correction and more of an FYI:

While this form of making a frequency dictionary is super clear, we have a one line way of doing this as well that you are welcome to use elsewhere! It might look something like the following:

```suggestion
    for letter in uppercase_word:
        check_word[letter] = check_word.get(letter, 0) + 1
```
On the right side of the assignment, we use the `.get(letter, 0)` function which not only gets the value associated with a key, but it provides a default of 0 if the key doesn't exist! Then we add one and assign it to that letter's key in the dictionary! Like I said, your way works well and is very clear, this is just an alternate approach that could be useful in future projects!